### PR TITLE
Fix mentor dashboard money calculation and remove methods no longer used

### PR DESCRIPTION
**Pull request by emerencia**
# Description of feature
...

# Checklist
- [x] (If applicable) added example values of new ENV variables to .env.local and explained them in README.md.
- [x] Added unit tests to test the code.
- [x] Added integration tests to test the code within the application as a whole.

### CODE REVIEW

File: app/models/reward.rb [hunk 1]
```diff
@@ -9,19 +9,26 @@ class Reward < ApplicationRecord
...
   CAN_STILL_BE_EARNED = 'can_still_be_earned'
 
-  def self.total_euros(bust_cache: false)
+  def self.total_earned_euros(bust_cache: false)
     RedisCachedCall.cache(TOTAL_EARNED_SO_FAR, bust_cache) do
       students = Person.all.reject(&:mentor?)
-      total_reward_points = students.sum(&:reward_points)
-      total_reward_points / 100.0
+      earned_reward_points = 0
+      students.each do |student|
```
Comment (frbl) [inline app/models/reward.rb]: `earned_reward_points.sum{|student| CalculateEarnedByPerson.run!(person: student)}` werkte ook toch?

File: app/models/reward.rb [hunk 2]
```diff
@@ -9,19 +9,26 @@ class Reward < ApplicationRecord
...
     end
   end
 
   def self.max_still_earnable_euros(bust_cache: false)
     RedisCachedCall.cache(CAN_STILL_BE_EARNED, bust_cache) do
       students = Person.all.reject(&:mentor?)
-      max_still_earnable_reward_points = students.sum(&:max_still_earnable_reward_points)
-      max_still_earnable_reward_points / 100.0
+      still_earnable_reward_points = 0
+      students.each do |student|
```
Comment (frbl) [inline app/models/reward.rb]: en hier ook

File: app/serializers/api/protocol_subscription_serializer.rb [hunk 3]
```diff
@@ -33,12 +33,14 @@ def max_still_awardable_euros
...
       latest_streak_value = completion[object.latest_streak_value_index]
       return 0 unless latest_streak_value.present?
       object.protocol.calculate_reward([latest_streak_value])
     end
 
     def current_multiplier
+      return 1 if no_streak_detected
       current_completion = completion[object.latest_streak_value_index]
       return 1 unless current_completion.present?
       latest_streak_value = completion[object.latest_streak_value_index][:streak]
```
Comment (frbl) [inline app/serializers/api/protocol_subscription_serializer.rb]: waarom hier geen `current_completion` btw? :p

File: app/use_cases/calculate_total_by_person.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class CalculateTotalByPerson < ActiveInteraction::Base
+  object :person
+
+  # Calculates the amount of money a person has earned
```
Comment (frbl) [inline app/use_cases/calculate_total_by_person.rb]: volgens mij klopt dit niet helemaal?

File: app/use_cases/calculate_total_by_person.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class CalculateTotalByPerson < ActiveInteraction::Base
```
Comment (frbl) [inline app/use_cases/calculate_total_by_person.rb]: `CalculateMaximumEurosByPerson`?

File: app/models/reward.rb (hunk 1 above)
Comment (emerencia) [reply app/models/reward.rb]: Wat hip. Fixed.

File: app/models/reward.rb (hunk 2 above)
Comment (emerencia) [reply app/models/reward.rb]: Fixed.

File: app/serializers/api/protocol_subscription_serializer.rb (hunk 3 above)
Comment (emerencia) [reply app/serializers/api/protocol_subscription_serializer.rb]: Dat is jouw code. Ik pas het wel aan.

File: app/use_cases/calculate_total_by_person.rb (hunk 4 above)
Comment (emerencia) [reply app/use_cases/calculate_total_by_person.rb]: Fixed.

**Pull request merged**